src/NuGet.Clients/NuGet.MSSigning.Extensions/MSSignCommand.cs

```diff
@@ -0,0 +1,224 @@
+
+            if (matchingCertCollection == null || matchingCertCollection.Count == 0)
+            {
+                throw new InvalidOperationException(NuGetMSSignCommand.MSSignCommandNoCertException);
+            }
+
+            return matchingCertCollection[0];
+        }
+
+        private void ValidatePackagePath()
+        {
+            // Assert mandatory argument
+            if (Arguments.Count < 1 ||
+                string.IsNullOrEmpty(Arguments[0]))
+            {
+                throw new ArgumentException(NuGetMSSignCommand.MSSignCommandNoPackageException);
+            }
+        }
+
+        private void WarnIfNoTimestamper(ILogger logger)
+        {
+            if (string.IsNullOrEmpty(Timestamper))
+            {
+                logger.Log(LogMessage.CreateWarning(NuGetLogCode.NU3521, NuGetMSSignCommand.MSSignCommandNoTimestamperWarning));
+            }
+        }
+
+        private void ValidateOutputDirectory()
+        {
+            if (!string.IsNullOrEmpty(OutputDirectory) &&
+                            !Directory.Exists(OutputDirectory))
+            {
+                Directory.CreateDirectory(OutputDirectory);
+            }
+        }
+
+        private void ValidateCertificateInputs()
+        {
+            if (string.IsNullOrEmpty(CertificateFile)
+                && Path.GetExtension(CertificateFile).Equals(".p7b", StringComparison.InvariantCultureIgnoreCase))
+            {
+                // Throw if user gave no certificate input
+                throw new ArgumentException(NuGetMSSignCommand.MSSignCommandNoValidCertificateFileException);
+            }
+
+            if (string.IsNullOrEmpty(CSPName))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                        NuGetMSSignCommand.MSSignCommandInvalidArgumentException,
+                        nameof(CSPName)));
+            }
+
+            if (string.IsNullOrEmpty(KeyContainer))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                        NuGetMSSignCommand.MSSignCommandInvalidArgumentException,
+                        nameof(KeyContainer)));
+            }
+
+            if (string.IsNullOrEmpty(CertificateFingerprint))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                        NuGetMSSignCommand.MSSignCommandInvalidArgumentException,
+                        nameof(CertificateFingerprint)));
+            }
+        }
+
+        private Common.HashAlgorithmName ValidateAndParseHashAlgorithm(string value, string name, SigningSpecifications spec)
+        {
+            var hashAlgorithm = Common.HashAlgorithmName.SHA256;
+
+            if (!string.IsNullOrEmpty(value))
+            {
+                if (!spec.AllowedHashAlgorithms.Contains(value, StringComparer.InvariantCultureIgnoreCase) ||
```

Comment:
Use `CryptoHashUtility` if possible, or add a common helper to parse this. Culture seems incorrect here